src/machine/machine-varlink.c

```diff
@@ -187,6 +188,9 @@ int vl_method_register(sd_varlink *link, sd_json_variant *parameters, sd_varlink
         if (r < 0)
                 return r;
 
+        if (!isempty(machine->root_directory) && path_is_os_tree(machine->root_directory) <= 0)
+                return sd_varlink_error(link, SD_VARLINK_ERROR_INVALID_PARAMETER, NULL);
```

Comment:
hmm, the assumption we can access that in full might not be true (i.e. because container uses an overlayfs instance on top of some incomplete disk image or so, but that overlayfs is only visible in the namespace of the container).

I don't think we should make this restriction here. Can you explain why you added this specifically?

i think we should simply refuse the individual operations on root directories if user cannot be trusted. but afaics bus_machine_method_open_root_directory() is not accessible unpriv at all right now so there is no issue here?

Reply:
ah, wait, not even in machine_open_root_directory() we actually use root_directory. WE enter the namespace and get the root fd direcly from there, which is the right way.

Hence, *already* the field has no relevance besides being informational. hence i don't think we need to try to authenticate it, it's never gonna work anyway (as you don't cover the root image stuff anyway).

Reply:
> Hence, already the field has no relevance besides being informational.

Ok we need to update the documentation then, as it's not very obvious it has no relevance